src/DynamoCoreWpf/Controls/NodeAutoCompleteSearchControl.xaml.cs

```diff
@@ -145,10 +145,13 @@ private void UnsubscribeFromAppEvents()
         //Hide the window and unsubscribe from model events.
         //Note that the window is not destroyed, it is just hidden so that it can be reused.
         internal void OnHideNodeAutoCompleteSearch()
-        {            
-            ViewModel.ParentNodeRemoved -= OnParentNodeRemoved;
-            ViewModel.OnNodeAutoCompleteWindowClosed();
-            Hide();
+        {
+            if (IsVisible) //when window is already hidden it means unsubscribe was already performed
```

Comment:
This particular change is not related to the PR subject but it is small and did not want to create another task for it.